src/ceph-disk/ceph_disk/main.py

```diff
@@ -2539,8 +2539,8 @@ def set_data_partition(self):
                       self.args.data)
             self.partition = DevicePartition.factory(
                 path=None, dev=self.args.data, args=self.args)
-            ptype = partition.get_ptype()
-            ready = Ptype.get_ready_by_type('osd')
+            ptype = self.partition.get_ptype()
+            ready = Ptype.get_ready_by_name('osd')
```

Comment:
good catch ! http://tracker.ceph.com/issues/15450 meaning there is no coverage for this code path